torch/distributed/_tensor/experimental/local_map.py

```diff
@@ -0,0 +1,174 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates
+from typing import Callable, Optional, Sequence, Tuple, Union
+
+import torch
+from torch.distributed._tensor import DeviceMesh, DTensor
+from torch.distributed._tensor.placement_types import Placement
+from torch.utils._pytree import tree_flatten, tree_unflatten
+
+
+PlacementType = Optional[Sequence[Placement]]
+InputPlacements = Union[PlacementType, Tuple[PlacementType, ...]]
+OutputPlacements = Union[PlacementType, Tuple[PlacementType, ...]]
+
+
+def local_map(
```

Comment:
I think we can have a better function name once the before the release to prod.

Reply:
any recommendation on the naming? ;-)